pwndbg/ui.py

```diff
@@ -38,9 +38,28 @@ def check_title_position():
         title_position.revert_default()
 
 
-def banner(title):
-    title = title.upper()
+def banner(title, make_upper=True, trunc_after_idx=-1):
+    """
+    Renders a banner.
+
+    Arguments:
+        title(str): banner title string
+        make_upper(bool): whether the title be uppercase
+        trunc_after_idx(int): if the value is bigger than 0,
+            and the title is too long, it will be truncated
+            on the left side but after the provided index
+    """
+    if make_upper:
+        title = title.upper()
+
     _height, width = get_window_size()
+
+    if trunc_after_idx > 0:
+        size_to_trunc = len(title) - (width - len('-[  ]-'))
```

Comment:
you can't use `'-[  ]-'` it can be overriden by `config.banner_title_surrounding_left` and `config.banner_title_surrounding_right` like below